optional_plugins/varianter_yaml_to_mux/tests/test_unit.py

```diff
@@ -11,8 +11,8 @@
 from avocado.utils import astring
 from avocado_varianter_yaml_to_mux import mux
 
-BASEDIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..')
-BASEDIR = os.path.abspath(BASEDIR)
+BASEDIR = os.path.dirname(os.path.abspath(__file__))
+BASEDIR = os.path.abspath(os.path.join(BASEDIR, os.path.pardir))
```

Comment:
Unless I'm missing something, in line 14 you are using `os.path.abspath` that will return already an absolute path. os.path.dirname will keep the absolute path. IMO, the second `abspath` (line 15) is unecessary.

What do you think?

Reply:
Hi @beraldoleal, that's a good question. From what I understand, the second `abspath` is to evaluate the absolute path after joining with `'..'` (or other O.S. equivalent). For instance, `os.path.join` would return something like `/home/user/avocado/optional_plugins/varianter_yaml_to_mux/tests/..` then, after applying `os.path.abspath` it would return `/home/user/avocado/optional_plugins/varianter_yaml_to_mux`.

Let me know if that answers your question.

Reply:
I got your point/example and now I can see why is there. Thanks. But yes, I think that maybe could improve readability a little bit:
```python
from os.path import abspath, dirname

BASEDIR = dirname(dirname(abspath(__file__))))
```

Not sure if will works, didn't tested. What do you think ?

Reply:
Considering we are using `os.path.` everywhere in the project, I don't see a problem using it here. I think importing and using the would confuse more based on what we already do in the project.

This does not mean we can not open an issue to discuss if we should move all the `os.path.` entries to specific functions.

Reply:
You have a point, I'm not going to block the PR because of this, but to be honest I don't like much the way it is. We need to get "the parent of the parent" and we using 2 lines with 4 "methods".

Reply:
Yep, I agree if you!

Reply:
> I got your point/example and now I can see why is there. Thanks. But yes, I think that maybe could improve readability a little bit:
> 
> ```python
> from os.path import abspath, dirname
> 
> BASEDIR = dirname(dirname(abspath(__file__))))
> ```
> 
> Not sure if will works, didn't tested. What do you think ?

This seems to return the same result from the initial testing I've done. I would say this would be more readable, even without importing `abspath` and `dirname` separately. 

> Considering we are using `os.path.` everywhere in the project, I don't see a problem using it here. I think importing and using the single funciton would confuse more based on what we already do in the project.
> 
> This does not mean we can not open an issue to discuss if we should move all the `os.path.` entries to specific functions.

That's a good point indeed. Another alternative would be to use `pathlib` alongside `os.path`, but this might be a bit confusing as well, what do you think @beraldoleal and @willianrampazzo? So far, the following seems to have a similar result:

```python
import pathlib

BASEDIR = pathlib.Path(__file__).resolve().parents[1]
```

> You have a point, I'm not going to block the PR because of this, but to be honest I don't like much the way it is. We need to get "the parent of the parent" and we are using 2 lines with 4 "methods".

I feel the same. I can open an issue if you all agree...

Reply:
> > Considering we are using `os.path.` everywhere in the project, I don't see a problem using it here. I think importing and using the single funciton would confuse more based on what we already do in the project.
> > This does not mean we can not open an issue to discuss if we should move all the `os.path.` entries to specific functions.
> 
> That's a good point indeed. Another alternative would be to use `pathlib` alongside `os.path`, but this might be a bit confusing as well, what do you think @beraldoleal and @willianrampazzo? So far, the following seems to have a similar result:
> 
> ```python
> import pathlib
> 
> BASEDIR = pathlib.Path(__file__).resolve().parents[1]
> ```
> 

We already use `os.path` in the project. Having another package to solve the same already solved by `os.path` seems to me just transferring the action from one package to another.

> > You have a point, I'm not going to block the PR because of this, but to be honest I don't like much the way it is. We need to get "the parent of the parent" and we are using 2 lines with 4 "methods".
> 
> I feel the same. I can open an issue if you all agree...

Discussion is always good to document the pros and cons of some approaches. If we decide what we have now is good enough, we can document that and close the issue.